version: "3.8"

services:
  server:
    build:
      context: ./server
      dockerfile: Dockerfile
    env_file:
      - ./server/.env
    ports:
      - "8761:3000"
    depends_on:
      postgres:
        condition: service_healthy
    volumes:
      - ./server:/app
      - /app/node_modules
    networks:
      - invoice_demo_network

  client:
    build:
      context: ./client
      dockerfile: Dockerfile
    env_file:
      - ./client/.env
    ports:
      - "8762:5173"
    depends_on:
      - server
    volumes:
      - ./client:/app
      - /app/node_modules
    networks:
      - invoice_demo_network

  postgres:
    image: postgres:16
    container_name: postgres_db
    restart: always
    environment:
      POSTGRES_USER: postgres_dbu
      POSTGRES_PASSWORD: invoice_db_pass
      POSTGRES_DB: invoice_dbn
    ports:
      - "8763:5432"
    volumes:
      - pgdata:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U user"]
      interval: 10s
      retries: 5
    networks:
      - invoice_demo_network

volumes:
  pgdata:

networks:
  invoice_demo_network:
    driver: bridge
